tools/chpldef/protocol-types.cpp

```diff
@@ -21,13 +21,43 @@
 #include "./protocol-types.h"
 #include "./misc.h"
 #include "llvm/Support/JSON.h"
+#include <cstdlib>
+#include <iostream>
+
+template <typename T>
+static void doAddField(chpldef::JsonObject& obj, const char* name,
+                       const T& field) {
+  obj[name] = field;
+}
+
+template <typename T>
+static void doAddField(chpldef::JsonObject& obj, const char* name,
```

Comment:
```suggestion
static void doAddField<chpldef::opt<T>>(chpldef::JsonObject& obj, const char* name,
```
If it works the way I think it should work...